### PR TITLE
Add generate command

### PR DESCRIPTION
**Pull request by arturdm**

### CODE REVIEW

File: test/command/generate_command_integration_test.dart [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        "build/tests.output",
+        "--coverage-path",
+        "build/lcov.info",
+        "${testDirectory}/build/sonar-project.properties",
+      ]);
+
+      // then
+      final actualFile = File("${testDirectory}/build/sonar-project.properties");
+      expect(actualFile.existsSync(), isTrue);
+      expect(actualFile.readAsStringSync(), isNotEmpty);
```
Comment (WezSieTato) [inline test/command/generate_command_integration_test.dart:39]: Maybe we could equal this file to proper output file saved? Something like snapshot, but for text file.

File: lib/src/module_finder/module_finder.dart [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  @override
+  List<Object> get props => [
+        modules,
+      ];
+
+  @override
+  bool get stringify => true;
+}
+
+class SonarModule extends Equatable {
```
Comment (WezSieTato) [inline lib/src/module_finder/module_finder.dart]: `SonarModule` in separate file?

File: lib/src/command/sonarscanner_command.dart [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import 'package:args/command_runner.dart';
+import 'package:moronepo/moronepo.dart';
+
+abstract class SonarScannerCommand<T> extends Command<T> {
```
Comment (WezSieTato) [inline lib/src/command/sonarscanner_command.dart:4]: Just asking: are you think about more commands than generate?

File: lib/src/module_finder/module_finder.dart [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    return file.existsSync() ? path : null;
+  }
+
+  String _relativePathToDirectoryIfExists(String base, String path) {
+    final directory = Directory("$base/$path");
+    return directory.existsSync() ? path : null;
+  }
+}
+
+class FindModuleResult extends Equatable {
```
Comment (WezSieTato) [inline lib/src/module_finder/module_finder.dart]: What is the purpose of this class?

**Review by WezSieTato — CHANGES_REQUESTED** [review]

File: lib/src/command/sonarscanner_command.dart (hunk 3 above)
Comment (arturdm) [reply lib/src/command/sonarscanner_command.dart:4]: PR coming up - run command to download and run `sonar-scanner` https://docs.sonarqube.org/latest/analysis/scan/sonarscanner/

File: lib/src/module_finder/module_finder.dart (hunk 4 above)
Comment (arturdm) [reply lib/src/module_finder/module_finder.dart]: I didn't want to return `Future<Iterable<SonarModule>>` from `findModules` method but I guess I can refactor it.

File: test/command/generate_command_integration_test.dart (hunk 1 above)
Comment (arturdm) [reply test/command/generate_command_integration_test.dart:39]: The file contains absolute paths :(

**Review by WezSieTato — APPROVED** [review]

**Pull request merged**